pkg/storegateway/bucket_stores.go

```diff
@@ -235,7 +235,7 @@ func (u *BucketStores) syncUsersBlocks(ctx context.Context, f func(context.Conte
 				if err := f(ctx, job.store); err != nil {
 					if errors.Is(err, bucket.ErrCustomerManagedKeyAccessDenied) {
 						u.storesErrorsMu.Lock()
-						u.storesErrors[job.userID] = err
+						u.storesErrors[job.userID] = httpgrpc.Errorf(int(codes.PermissionDenied), "store error: %s", err)
```

Comment:
Any reason we wrap this to httpgrpc error?

Reply:
Seems it is used in testing only?

Reply:
This get translated to a 4xx here:

https://github.com/cortexproject/cortex/blob/1d409248a9d1c1e959de5ffaad0c9070372cdc52/pkg/querier/blocks_store_queryable.go#L653-L655

This was causing 5xx in this case.